### ISSUE TITLE
Support for fact-oriented map in clara.tools.inspect/inspect

### ISSUE THREAD

**Issue by WilliamParker** [rbrush/clara-rules#161]
I have found the ability to quickly determine what facts were inserted by a particular rule to be quite helpful in debugging rules projects.  However, while it includes all necessary data, it is awkward to use a map from rules to insertions when one has a particular fact and wants to determine what rule inserted that fact.  I believe it would be helpful to have an additional key of the data map returned by inspect contain a map from inserted facts to the rule that inserted them.

**Comment by WilliamParker** [comment]
I have logged a pull request for this at https://github.com/rbrush/clara-rules/pull/162

I decided on a structure of {fact-inserted [rule-and-explanation-map]}, with multiple entries for duplicate facts in the session.  I considered having [rule-and-explanation] be a frequencies map instead (i.e. {rule-and-explanation number-of-occurences} but that seemed like it added too much complication for not enough benefit.

Please let me know if you have any other thoughts, suggestions, etc.  I updated the docstring for inspect; I don't know if there is any other documentation on inspection that would need to be updated if this change is made.

**Comment by rbrush** [comment]
I like this approach, and the map of facts to rule+explanation makes sense. I made a comment over on the pull request itself but will merge this once that lands.

**Comment by WilliamParker** [comment]
I submitted a new pull request with the changes at https://github.com/rbrush/clara-rules/pull/163

**Comment by rbrush** [comment]
Merged. Thanks for the contribution!

(For future reference, I don't mind squashing/replacing an existing pull request but creating a new one works too.)

**Comment by rbrush** [comment]
Closing this since I don't think there is anything left to do here. We can reopen if I'm mistaken.
